packages/principal/test/ed25519.spec.js

```diff
@@ -0,0 +1,181 @@
+import { ed25519 as Lib } from '../src/lib.js'
+import { assert } from 'chai'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { varint } from 'multiformats'
+
+describe('signing principal', () => {
+  const { Signer } = Lib
+
+  it('exports', () => {
+    assert.equal(Lib.code, 0x1300)
+    assert.equal(Lib.name, 'Ed25519')
+    assert.equal(Lib.signatureAlgorithm, 'EdDSA')
+    assert.equal(Lib.signatureCode, 0xd0ed)
+    assert.equal(typeof Lib.derive, 'function')
+    assert.equal(typeof Lib.generate, 'function')
+
+    assert.equal(typeof Lib.Verifier, 'object')
+    assert.equal(typeof Lib.Signer, 'object')
+  })
+
+  it('generate', async () => {
+    const signer = await Lib.generate()
+    assert.ok(signer.did().startsWith('did:key'))
+    assert.equal(signer.code, 0x1300)
+    assert.equal(signer.signatureCode, 0xd0ed)
+    assert.equal(signer.signatureAlgorithm, 'EdDSA')
+    assert.equal(signer.signer, signer)
+    assert.equal(signer.verifier.code, 0xed)
+    assert.equal(signer.verifier.signatureCode, 0xd0ed)
+    assert.equal(signer.verifier.signatureAlgorithm, 'EdDSA')
+
+    const payload = await sha256.encode(new TextEncoder().encode('hello world'))
+    const signature = await signer.sign(payload)
+
+    const verifier = Lib.Verifier.parse(signer.did())
+    assert.equal(
+      await verifier.verify(payload, signature),
+      true,
+      'signer can verify signature'
+    )
+    assert.equal(await signer.verify(payload, signature), true)
+
+    assert.equal(signer.signatureAlgorithm, 'EdDSA')
+    assert.equal(signer.signatureCode, 0xd0ed)
+  })
+
+  it('derive', async () => {
+    const original = await Lib.generate()
+    // @ts-expect-error - secret is not defined by interface
+    const derived = await Lib.derive(original.secret)
```

Comment:
ed has a derive method ? we need to have a smaller api surface please and consistency in the names what ever they are